integration/test/app/basic/simple_test.go

```diff
@@ -0,0 +1,199 @@
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		if !reflect.DeepEqual(chart.Spec.TarballURL, tarballURL) {
+			t.Fatalf("expected tarballURL: %#v got %#v", tarballURL, chart.Spec.TarballURL)
+		}
+		originalResourceVersion = chart.ObjectMeta.ResourceVersion
+	}
+
+	// Test update
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating app %#q", customResourceReleaseName))
+
+		sampleChart.App.Version = "1.0.1"
+		sampleChart.AppCatalog.Storage.URL = "https://giantswarm.github.com/sample-catalog_1/"
+
+		chartValues, err := chartvalues.NewAPIExtensionsAppE2E(sampleChart)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		chartInfo := release.NewStableChartInfo(customResourceReleaseName)
+		err = config.Release.Update(ctx, customResourceReleaseName, chartInfo, chartValues)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		err = config.Release.WaitForStatus(ctx, fmt.Sprintf("%s-%s", namespace, customResourceReleaseName), "DEPLOYED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updated app %#q", customResourceReleaseName))
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chart CR %#q is updated", testAppReleaseName))
+
+		tarballURL := "https://giantswarm.github.com/sample-catalog_1/test-app-1.0.1.tgz"
+		err = waitForChartUpdated(ctx, update, originalResourceVersion)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		chart, err := config.Host.G8sClient().ApplicationV1alpha1().Charts(namespace).Get(testAppReleaseName, v1.GetOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		if !reflect.DeepEqual(chart.Spec.TarballURL, tarballURL) {
+			t.Fatalf("expected tarballURL: %#v got %#v", tarballURL, chart.Spec.TarballURL)
+		}
+	}
+
+	// Test deletion
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting app %#q", customResourceReleaseName))
+
+		err := config.Release.Delete(ctx, customResourceReleaseName)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		err = config.Release.WaitForStatus(ctx, fmt.Sprintf("%s-%s", namespace, customResourceReleaseName), "DELETED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted app %#q", customResourceReleaseName))
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chart CR %#q is deleted", testAppReleaseName))
+
+		err = waitForChartUpdated(ctx, delete, "")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+	}
+}
+
+// searchChart will find Chart CR which have name as testAppReleaseName and resourceVersion greater than one we have.
```

Comment:
Comment needs updating. What about using `waitForUpdatedChartCR` as the name? That is clearer for me.